test/testdata/lsp/fast_path/static_field_loc.rb

```diff
@@ -0,0 +1,27 @@
+# typed: true
+
+# This is a rather large comment at the start of the file so that when we
+# delete it on the fast path, it will make it very likely that we have an array
+# access that's out of bounds when reporting errors.
+# This is a rather large comment at the start of the file so that when we
+# delete it on the fast path, it will make it very likely that we have an array
+# access that's out of bounds when reporting errors.
+# This is a rather large comment at the start of the file so that when we
+# delete it on the fast path, it will make it very likely that we have an array
+# access that's out of bounds when reporting errors.
+# This is a rather large comment at the start of the file so that when we
+# delete it on the fast path, it will make it very likely that we have an array
+# access that's out of bounds when reporting errors.
+# This is a rather large comment at the start of the file so that when we
+# delete it on the fast path, it will make it very likely that we have an array
+# access that's out of bounds when reporting errors.
+# This is a rather large comment at the start of the file so that when we
+# delete it on the fast path, it will make it very likely that we have an array
+# access that's out of bounds when reporting errors.
+#
+class B
+  extend T::Generic
+  R = 1
+  R = type_member
```

Comment:
This is not a static field, this is a type_member.

I believe the crash reproduces even if the test does not have `R` defined simultaneously as a static field and type member.

Can you simplify this test to delete `type_member`, and deal with type members either in a follow up commit on this PR, or in a future PR?